.github/workflows/check-whitespace.yml

```diff
@@ -0,0 +1,71 @@
+name: check-whitespace
+
+# Get the repo with the commits(+1) in the series.
+# Process `git log --check` output to extract just the check errors.
+# Add a comment to the pull request with the check errors.
+
+on:
+  pull_request:
+    types: [opened, synchronize]
+
+jobs:
+  check-whitespace:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Set commit count
+      shell: bash
+      run: echo "::set-env name=CW_COMMITS::$((1+$COMMITS))"
+      env:
+        COMMITS: ${{ github.event.pull_request.commits }}
+
+    - uses: actions/checkout@v2
+      with:
+        fetch-depth: ${{ env.CW_COMMITS }}
+
+    - name: git log --check
+      id: check_out
+      run: |
+        set +e
+        log=
+        commit=
+        while read dash etc
+        do
+          case "${dash}" in
+          "---")
+            commit="${etc}"
+            ;;
+          "")
+            ;;
+          *)
+            if test -n "${commit}"
+            then
+              log="${log}\n${commi}t"
+              echo ""
+              echo "--- ${commit}"
+            fi
+            commit=
+            log="${log}\n${dash} ${etc}"
+            echo "${dash} ${etc}"
+            ;;
+          esac
+        done <<< $(git log --check --pretty=format:"---% h% s" -${{github.event.pull_request.commits}})
```

Comment:
I believe that a `sed` script would be preferable. Something like this:

```sh
git log --check --pretty=format:"---% h% s" -${{github.event.pull_request.commits}} |
sed '/^--- /{
  :1
  s/.*\n//
  N
  /\n--- /b1
  /\n$/b1
}' |
sed '${/^--- /d}'
```

Also, I am sure that the output can be made prettier in the subsequent step, e.g. by enclosing the per-commit metadata into backticked blocks.

Reply:
> I believe that a `sed` script would be preferable. Something like this:

Not sure how a `sed` script with loops is cleaner than the shell script.  The string `\n` is part of the final string, replacing real line ends, so that also has to be accounted for.  Not sure how real line ends are treated in environment variables.

> Also, I am sure that the output can be made prettier in the subsequent step, e.g. by enclosing the per-commit metadata into backticked blocks.

GitHub hyperlinks the commits (see [test PR](https://github.com/webstech/gitout/pull/170#issuecomment-683907287)).  Backticks on the commit line voids the hyperlinking, so we don't want to do that.  Turning the whitespace errors into hyperlinks may be possible but I'll leave that for a future patch.